stack/tools/src/main/java/org/apache/usergrid/tools/ImportAdmins.java

```diff
@@ -133,84 +163,129 @@ private void importAdminUsers() throws Exception {
+            if (loopCounter % 100 == 1)
+                logger.info("Publishing to queue... counter=" + loopCounter);
 
-            @SuppressWarnings( "unchecked" )
-            Map<String, Object> entityProps = jp.readValueAs( HashMap.class );
+            workQueue.add(entityProps);
+        }
 
-            // Import/create the entity
-            UUID uuid = getId( entityProps );
-            String type = getType( entityProps );
+        waitForQueueAndMeasure(workQueue, adminWriteThreads, "Admin Write");
+        waitForQueueAndMeasure(auditQueue, adminAuditThreads, "Admin Audit");
 
+        logger.info("----- End: Imported {} admin users from file {}",
+                count, adminUsersFile.getAbsolutePath());
 
-            try {
-                em.create( uuid, type, entityProps );
+        jp.close();
+    }
 
-                logger.debug( "Imported admin user {} {}", uuid, entityProps.get( "username" ) );
-                count++;
-                if ( count % 1000 == 0 ) {
-                    logger.info("Imported {} admin users", count);
-                }
-            }
-            catch ( DuplicateUniquePropertyExistsException de ) {
-                logger.warn( "Unable to create entity. It appears to be a duplicate: " +
-                    "id={}, type={}, name={}, username={}",
-                    new Object[] { uuid, type, entityProps.get("name"), entityProps.get("username")});
-                if ( logger.isDebugEnabled() ) {
-                    logger.debug( "Exception" , de );
-                }
-                continue;
-            }
+    private static void waitForQueueAndMeasure(final BlockingQueue workQueue,
+                                               final Map<Stoppable, Thread> threadMap,
+                                               final String identifier) throws InterruptedException {
+        double rateAverageSum = 0;
+        int iterationCounter = 0;
 
-            if ( em.get( uuid ) == null ) {
-                logger.error( "Holy hell, we wrote an entity and it's missing.  " +
-                                "Entity Id was {} and type is {}", uuid, type );
-                System.exit( 1 );
-            }
-            echo( entityProps );
+        while (!workQueue.isEmpty()) {
+            iterationCounter += 1;
+
+            int sizeLast = workQueue.size();
+            long lastTime = System.currentTimeMillis();
+            logger.info("Queue {} is not empty, remaining size={}, waiting...", identifier, sizeLast);
+            Thread.sleep(5000);
+
+            long timeNow = System.currentTimeMillis();
+            int sizeNow = workQueue.size();
+
+            int processed = sizeLast - sizeNow;
+
+            long timeDelta = timeNow - lastTime;
+
+            double rateLast = (double) processed / (timeDelta / 1000);
+            rateAverageSum += rateLast;
+
+            long timeRemaining = sizeLast / (long) (rateAverageSum / iterationCounter);
```

Comment:
I had to change this to:

long timeRemaining = (long) ( sizeLast / (rateAverageSum / iterationCounter) );

Otherwise the unit tests throws a divide by zero error.